plugins/redis/provisioner.go

```diff
@@ -0,0 +1,44 @@
+package redis
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+var argsToProvision = []string{
+	"--user", fieldname.Username.String(),
+	"-h", fieldname.Host.String(),
+	"-p", fieldname.Port.String(),
+}
+
+type redisArgsProvisioner struct {
+}
+
+func redisProvisioner() sdk.Provisioner {
+	return redisArgsProvisioner{}
+}
+
+func (p redisArgsProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	if value, ok := in.ItemFields[fieldname.Password]; ok {
+		out.AddEnvVar("REDISCLI_AUTH", value)
+	}
+	for i, arg := range argsToProvision {
+		if i%2 == 0 {
+			argName := arg
+			fieldName := sdk.FieldName(argsToProvision[i+1])
+			if fieldValue, ok := in.ItemFields[fieldName]; ok {
+				out.AddArgsAtIndex(1, argName, fieldValue)
+			}
```

Comment:
I'm still convinced that this is the easier way to write this, as I need to spend some time figuring out what this does. 

Why don't we just use the following approach? We don't need to use an array because we only have 3 values we are interested in.
```suggestion
	if fieldValue, ok := in.ItemFields[fieldname.Username]; ok && CommandLineDoesNotContainUser {
		out.AddArgsAtIndex(1, "--user", fieldValue)
	}
	if fieldValue, ok := in.ItemFields[fieldname.Host]; ok && CommandDoesNotContainHost{
		out.AddArgsAtIndex(1, "-h", fieldValue)
	}
	if fieldValue, ok := in.ItemFields[fieldname.Port]; ok && CommandDoesNotContainPort {
		out.AddArgsAtIndex(1, "-p", fieldValue)
	}
```

Reply:
Or maybe just set the command line directly? Something like:

```go
commandLine := []string{out.CommandLine[0], "--user", user, "--host", host, "--port", port}
commandLine = append(commandLine, out.CommandLine[1:]...)
out.CommandLine = commandLine
```